cmake/Testing.cmake

```diff
@@ -66,6 +66,51 @@ function(openassetio_add_generic_test_target target_name)
     )
 endfunction()
 
+#-----------------------------------------------------------------------
+# ABI checker tests.
+
+if (OPENASSETIO_ENABLE_TEST_ABI)
+    find_program(OPENASSETIO_ABIDIFF_EXE NAMES abidiff REQUIRED)
+
+    # Create a test to check the ABI compatibility of a target.
+    #
+    # Uses libabigail's `abidiff` tool, and assumes .xml snapshots with
+    # the same name as the library, under `resources/abi` in the
+    # project root.
+    #
+    # Uses the build artifact location for convenience (i.e. handy
+    # generator expressions to get the path), rather than the install
+    # artifacts, but they are equivalent for `abidiff` purposes.
+    function(openassetio_add_abi_test_target target_name)
+        add_custom_target(
+            openassetio.test.abi.${target_name}
+            COMMAND ${CMAKE_COMMAND} -E echo
```

Comment:
[Consider][Minor] I think most people wouldn't realise this is a newline, maybe a comment.